cfme/tests/cloud/test_providers.py

```diff
@@ -597,11 +598,38 @@ def test_refresh_with_empty_iot_hub_azure(request, provider, setup_provider):
     assert result.validate(wait="60s")
 
 
+@pytest.fixture
+def azure_available_regions(provider):
+    def _func():
+        view = navigate_to(AzureProvider, "Add")
+        view.browser.refresh()
+        # prefill the provider type to enable regions dropdown
+        view.fill({'prov_type': provider.type.capitalize()})
+        return [opt.text for opt in view.region.all_options][1:]  # first option is '<Choose>'
+    return _func
+
+
+@pytest.fixture
+def azure_available_regions_backend(appliance):
+    regions_rb = appliance.ssh_client.run_command(
+        "cat /opt/rh/cfme-gemset/bundler/gems/cfme-providers-azure-*/"
+        "app/models/manageiq/providers/azure/regions.rb")
+    c = [x.group(1) for x in re.finditer(r':description => _\(\"([ \w]*)\"\)',
+                                         regions_rb.output,
+                                         flags=re.MULTILINE)]
+    return c
+
+
+@pytest.fixture
+def azure_has_available_gov_regions(azure_available_regions):
+    return lambda: any(reg for reg in azure_available_regions() if 'gov' in reg.lower())
```

Comment:
I don't think this should be a fixture.

It obfuscates the assertion content of the test, and moves reading of the list in the UI to test setup, instead of test execution.